src/jdk.crypto.ec/share/classes/sun/security/ec/ECPrivateKeyImpl.java

```diff
@@ -124,13 +125,15 @@ private void makeEncoding(BigInteger s) throws InvalidKeyException {
             int outPos = Math.max(sOctets.length - sArr.length, 0);
             int length = Math.min(sArr.length, sOctets.length);
             System.arraycopy(sArr, inPos, sOctets, outPos, length);
+            Arrays.fill(sArr, (byte)0);
 
             DerOutputStream out = new DerOutputStream();
             out.putInteger(1); // version 1
             out.putOctetString(sOctets);
-            DerValue val =
-                new DerValue(DerValue.tag_Sequence, out.toByteArray());
+            Arrays.fill(sOctets, (byte)0);
```

Comment:
The same handling should apply to line 106 above inside makeEncoding(byte[])?

Reply:
Same as above. Both `putInteger` and `putOctetString` calls `DerOutputStream::putLength`.

Reply:
I found out the reason. This method is called during key pair generation but my test only deals with manually crafted keys (so that I know what special bytes to search for). I think I'll need to find out a different test method. This might reveal other leaks in key generation.

Reply:
Yes, your approach is based on the test and the code path it exercised. I manually code inspected the files that you touched. Even with both approaches, it's still not gonna be 100%.